src/Control/Carrier/Interpret.hs

```diff
@@ -54,19 +56,19 @@ reify a k = unsafeCoerce (Magic k) a
 --
 -- @since 1.0.0.0
 runInterpret
-  :: (HFunctor eff, Monad m)
+  :: (Effect eff, Monad m)
   => (forall x . eff m x -> m x)
   -> (forall s . Reifies s (Handler eff m) => InterpretC s eff m a)
   -> m a
-runInterpret f m = reify (Handler (InterpretC . f . handleCoercible)) (go m) where
+runInterpret f m = reify (Handler (InterpretC . fmap runIdentity . f . handle (Identity ()) (fmap Identity . coerce))) (go m) where
```

Comment:
This is the other casualty of `handleCoercible`: we can’t use it here any more. I’m ok with that, again because this is something most users are unlikely to have to do themselves. In particular, users of `runInterpret` itself certainly don’t—`runInterpretState` bears no such scar.